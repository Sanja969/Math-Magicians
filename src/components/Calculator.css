.Calculator-box {
  margin-top: 200px;
  width: 90%;
}

button {
  height: 50px;
  border-color: #efefef !important;
  font-size: 24px;
}

.raw > button:last-child{
  background-color: #f5913e;
}

.Display {
  height: 50px;
  background-color: rgb(60, 60, 57);
  color: aliceblue;
}

@media (min-width: 768px) {
  .Calculator-box {
    width: 75%;
  }
  button {
    height: 100px;
  }
  .Display {
    height: 100px;
  }
}

@media (min-width: 1500px) {
  .Calculator-box {
    width: 50%;
  }
}