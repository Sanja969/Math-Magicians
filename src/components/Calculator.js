import React, { useState } from 'react';
import KeysRaw from './keys';
import './Calculator.css';
import calculate from '../logic/calculate';

const raw1 = ['AC', '+/-', '%', 'รท'];
const raw2 = ['7', '8', '9', 'x'];
const raw3 = ['4', '5', '6', '-'];
const raw4 = ['1', '2', '3', '+'];
const raw5 = ['0', '.', '='];

const Calculator = () => {
  const [obj, setObj] = useState({ total: '0', next: null, operation: null });
  const { total, next, operation } = obj;

  const handleClick = (value) => {
    setObj(calculate(obj, value));
  };

  return (
    <div className="Calculator-box w-75">
      <div className="Display d-flex align-items-center justify-content-end p-2 h2 m-0" title="display">
        {total}
        {operation}
        {next}
      </div>
      <div className="Calculator-body">
        <div className="raw">
          <KeysRaw raw={raw1} handleClick={handleClick} key={1} />
        </div>
        <div className="raw">
          <KeysRaw raw={raw2} handleClick={handleClick} key={2} />
        </div>
        <div className="raw">
          <KeysRaw raw={raw3} handleClick={handleClick} key={3} />
        </div>
        <div className="raw">
          <KeysRaw raw={raw4} handleClick={handleClick} key={4} />
        </div>
        <div className="raw">
          <KeysRaw raw={raw5} handleClick={handleClick} key={5} />
        </div>
      </div>
    </div>
  );
};

export default Calculator;
