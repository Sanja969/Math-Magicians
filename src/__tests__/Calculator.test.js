import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

const { container } = render(<Calculator />);
const display = container.querySelector('.Display');
const buttons = container.querySelectorAll('button');

describe('Test Calculator component', () => {
  it('makes operations correctly ', () => {
    buttons.forEach((button) => {
      if (button.textContent === '4') {
        fireEvent.click(button);
      }
    });
    buttons.forEach((button) => {
      if (button.textContent === '+') {
        fireEvent.click(button);
      }
    });
    buttons.forEach((button) => {
      if (button.textContent === '2') {
        fireEvent.click(button);
      }
    });
    buttons.forEach((button) => {
      if (button.textContent === '=') {
        fireEvent.click(button);
      }
    });
    expect(display.textContent).toBe('6');
  });
  it('renders correctly', () => {
    buttons.forEach((button) => {
      if (button.textContent === 'AC') {
        fireEvent.click(button);
        expect(button).not.toBeNull();
      }
    });
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
