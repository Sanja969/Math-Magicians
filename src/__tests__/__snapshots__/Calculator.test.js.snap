// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Calculator component matches snapshot 1`] = `
<div
  className="Calculator-box w-75"
>
  <div
    className="Display d-flex align-items-center justify-content-end p-2 h2 m-0"
    title="display"
  >
    0
  </div>
  <div
    className="Calculator-body"
  >
    <div
      className="raw"
    >
      <button
        className="col-3"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="col-3"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="col-3"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="col-3"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="raw"
    >
      <button
        className="col-3"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="col-3"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="col-3"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="col-3"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="raw"
    >
      <button
        className="col-3"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="col-3"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="col-3"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="col-3"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="raw"
    >
      <button
        className="col-3"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="col-3"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="col-3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="col-3"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="raw"
    >
      <button
        className="col-6"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="col-3"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="col-3"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
